x-pack/plugins/fleet/server/services/package_policies_to_agent_permissions.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { SavedObjectsClientContract } from 'kibana/server';
+
+import type { FullAgentPolicyOutputPermissions } from '../../common';
+import { getPackageInfo } from '../../server/services/epm/packages';
+
+import type { PackagePolicy } from '../types';
+
+export const DEFAULT_PERMISSIONS = {
+  cluster: ['monitor'],
+  indices: [
+    {
+      names: [
+        'logs-*',
+        'metrics-*',
+        'traces-*',
+        'synthetics-*',
+        '.logs-endpoint.diagnostic.collection-*',
+      ],
+      privileges: ['auto_configure', 'create_doc'],
+    },
+  ],
+};
+
+export async function storedPackagePoliciesToAgentPermissions(
+  soClient: SavedObjectsClientContract,
+  packagePolicies: string[] | PackagePolicy[]
+): Promise<FullAgentPolicyOutputPermissions | undefined> {
+  if (packagePolicies.length === 0) {
+    return;
+  }
+
+  // I'm not sure what permissions to return for this case, so let's return the defaults
+  if (typeof packagePolicies[0] === 'string') {
+    return { _fallback: DEFAULT_PERMISSIONS };
+  }
+
+  const permissionEntries = (packagePolicies as PackagePolicy[]).map<Promise<[string, any]>>(
+    async (packagePolicy) => {
+      if (!packagePolicy.package) {
+        return [packagePolicy.name, DEFAULT_PERMISSIONS];
+      }
+
+      const pkg = await getPackageInfo({
+        savedObjectsClient: soClient,
+        pkgName: packagePolicy.package.name,
+        pkgVersion: packagePolicy.package.version,
+      });
+
+      // FIXME is this true?
+      if (!pkg.data_streams) {
```

Comment:
If a package does not have any data streams, I would assume it doesn't need any permissions? For example fleet-server?

Reply:
That's my assumption too, yes